File: python/leetcode/battleships-in-a-board.py
class Solution:
    def countBattleships(self, board):
        """
        :type board: List[List[str]]
        :rtype: int
        """
        count = 0
        for c in range(len(board)):
            for r in range(len(board[0])):
                if board[c][r] == 'X' and (r - 1 < 0 or board[c][r - 1] != 'X') and (c - 1 < 0 or board[c - 1][r] != 'X'):
                    count += 1
        return count

File: java/leetcode/battleships-in-a-board.java
class Solution {
    public int countBattleships(char[][] board) {
        int count = 0;
        for (int c = 0; c < board.length; c++) {
            for (int r = 0; r < board[0].length; r++) {
                if (board[c][r] == 'X' && (r - 1 < 0 || board[c][r - 1] != 'X') && (c - 1 < 0 || board[c - 1][r] != 'X')) {
                    count++;
                }
            }
        }
        return count;
    }
}
